src/sema/mutability.rs

```diff
@@ -355,6 +355,11 @@ fn recurse_statements(stmts: &[Statement], ns: &Namespace, state: &mut StateChec
 
 fn read_expression(expr: &Expression, state: &mut StateCheck) -> bool {
     match expr {
+        Expression::StructMember { loc, expr, .. } if expr.ty().is_contract_storage() => {
+            state.data_account |= DataAccountUsage::READ;
+            state.read(loc);
+            return false;
+        }
```

Comment:
Why is this? Could do with a comment

Reply:
I'm confused, why should this match arm not be here? What should the comment say?

Reply:
Why should a struct member expression read from storage?

Reply:
I mean _storage_ structs read contract storage? 

` if expr.ty().is_contract_storage()`

Reply:
Just calculating the storage offset of a struct member does not access state

Reply:
See https://github.com/hyperledger/solang/blob/main/src/codegen/expression.rs#L648 
We just add the offset of the struct field to the storage key/offset

Reply:
Ok, I need to go over this again. Apparently, this is not just calculating the storage offset. It's actually accessing the field. Otherwise this fix wouldn't work. Thanks for pointing out